crypto/fipsmodule/hmac/hmac.c

```diff
@@ -583,6 +615,10 @@ int HMAC_Init_from_precomputed_key(HMAC_CTX *ctx,
   }
 
   const HmacMethods *methods = ctx->methods;
+  if (methods != NULL && ctx->methods->init_from_state == NULL) {
```

Comment:
I assume that `hmac_ctx_set_md_methods` is guaranteein that methods is not NULL, so you can just remove `methods != NULL` (if this is not the case, the existing code is already wrong as `methods->chaining_length` may crash).

Reply:
Yes, it looks like [that's ensured](https://github.com/aws/aws-lc/blob/587cf97c404a6c32f212e5c44ccb3503cb5ff430/crypto/fipsmodule/hmac/hmac.c#L335-L337), so this null check is redundant.